File: RhymingDictionary/src/tests/TestGetSounds.py
import unittest
from rhymes import RhymingDictionary


class TestGetSounds(unittest.TestCase):

    def setUp(self):
        pass

    def test_1(self):
        self.assertEqual(RhymingDictionary.getSounds("data/cmudict-0.7b", "CAR"), ['K', 'AA1', 'R'])

    def test_Shipping2(self):
        self.assertEqual(RhymingDictionary.getSounds("data/cmudict-0.7b", "FAR"), ['F', 'AA1', 'R'])

    def test_Shipping3(self):
        self.assertEqual(RhymingDictionary.getSounds("data/cmudict-0.7b", ""), [])


if __name__ == '__main__':
    unittest.main()

File: RhymingDictionary/src/tests/TestGetSounds.scala
package tests

import org.scalatest._
import rhymes.RhymingDictionary._

class TestGetSounds extends FunSuite {

  test("TestGetSounds"){
    val filename: String = "C:\\Users\\woodc\\IdeaProjects\\RhymingDictionaryHW1\\data\\cmudict-0.7b.txt"

    val test_a: String = "AND"
    val test_b: String = "HALF"

    assert(getSounds(filename,test_a) == List("AH0","N","D"))
    assert(getSounds(filename,test_b) == List("HH","AE1","F"))
    assert(getSounds(filename,"HBDFJSDBFSJD") == List())
  }
}